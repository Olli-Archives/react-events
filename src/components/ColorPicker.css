.ColorPicker{
  button{
    cursor:pointer
  }
  .red{
    background-color:red;
  }
  .blue{
    background-color:blue;
  }
  .green{
    background-color:green;
  }
}