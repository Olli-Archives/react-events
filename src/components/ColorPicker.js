import React, { PureComponent, fragment } from 'react';
import PropTypes from 'prop-types';
import styles from './ColorPicker.css';

export default class ColorPicker extends PureComponent {
  static propTypes = {
    color: PropTypes.string.isRequired
  }

  
  clickHandler =  ()=>{
    console.log('you clicked', this.props.color);
  }
  
  render(){
    const { color } = this.props;

    return (
      <section className={styles.ColorPicker}>
        <button className={styles[color]} onClick={this.clickHandler} >{this.props.color}</button> 
      </section>
    );
  }
}
