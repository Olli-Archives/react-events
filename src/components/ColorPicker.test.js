import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';


describe('ColorPicker component', ()=>{
  it('checks colorpicker snapshot', ()=>{
    const wrapper = shallow(<ColorPicker color='green'/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
