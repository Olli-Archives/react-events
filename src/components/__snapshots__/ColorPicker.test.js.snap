// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component checks colorpicker snapshot 1`] = `
<section
  className="ColorPicker"
>
  <button
    className="green"
    onClick={[Function]}
  >
    green
  </button>
</section>
`;
